pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -190,6 +189,21 @@ func (els *EventListenerStatus) SetCondition(newCond *apis.Condition) {
 	}
 }
 
+func (els *EventListenerStatus) MarkTrue() {
+	eventListenerCondSet.Manage(els).SetCondition(apis.Condition{
+		Type:   apis.ConditionReady,
+		Status: corev1.ConditionTrue,
+	})
+}
+
+// GetConditionSet retrieves the condition set for this resource. Implements
+// the KRShaped interface.
```

Comment:
Question: Do we have to implement the KRShaped interface as part of this PR? I think this adds a new `status.Condition` of type `Ready` -- which is what we'd want but we'd also have to add logic for when `Ready` is true (e.g. when deployment/service are ready).

Reply:
Pipeline seems to disable it for the existing types at the moment: https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L41

Reply:
yes a new condition of type `Ready` handled here https://github.com/tektoncd/triggers/pull/767/files#diff-ac6d17e220f6c870c0bc0304bc7c53b4R115-R117

Reply:
If we want to do it same like pipeline we can disable it on issues

Let me know

Reply:
Yeah, so if we are adding Ready we should ensure that it is only true iff both deployment and service are also ready. I think this will be a useful follow up. But since that's a feature and adds new fields to the type, we can add it in a follow up?

Reply:
Ya that make sense :+1: